### PR TITLE
Implement scheduler’s endpoint for getting the status of a restoration request

### PR DESCRIPTION
**Pull request by redouan-rhazouani**
Store status metadta in object store

PR is part of a series (#264) of PRs to implement distributed backup operations

### Review checklist

- [x] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [x] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: usecases/backup/coordinator.go [hunk 1]
```diff
@@ -188,30 +189,38 @@ func (c *coordinator) Restore(ctx context.Context, req *backup.DistributedBackup
...
+		filename = GlobalRestoreFile
 	}
-
 	// The backup might have been already created.
-	meta, err := store.Meta(ctx, req.ID)
+	meta, err := store.Meta(ctx, filename)
 	if err != nil {
 		path := fmt.Sprintf("%s/%s", req.ID, GlobalBackupFile)
-		return reqStat{}, fmt.Errorf("%w: %q: %v", errMetaNotFound, path, err)
+		return nil, fmt.Errorf("%w: %q: %v", errMetaNotFound, path, err)
```
Comment (parkerduckworth) [inline usecases/backup/coordinator.go:215]: Should the last `%v` should be a `%w`?

File: usecases/backup/scheduler.go [hunk 2]
```diff
@@ -104,35 +104,42 @@ func (s *Scheduler) Restore(ctx context.Context, pr *models.Principal,
...
-	return RestoreStatus{}, backup.NewErrUnprocessable(fmt.Errorf("not implemented"))
+func (s *Scheduler) RestorationStatus(ctx context.Context, principal *models.Principal, backend, backupID string,
+) (_ *Status, err error) {
+	path := fmt.Sprintf("backups/%s/%s/restore", backend, backupID)
+	if err := s.authorizer.Authorize(principal, "get", path); err != nil {
+		return nil, err
+	}
+	store, err := coordBackend(s.backends, backend, backupID)
+	if err != nil {
+		err = fmt.Errorf("no backup provider %q, did you enable the right module?", backend)
```
Comment (parkerduckworth) [inline usecases/backup/scheduler.go:134]: The original `err` is being swallowed here. Would it be better to include the original `err` in this reassignment?

File: usecases/backup/coordinator.go (hunk 1 above)
Comment (redouan-rhazouani) [reply usecases/backup/coordinator.go:215]: We can only use one %w in an expression.
The first one is the most important one

File: usecases/backup/scheduler.go (hunk 2 above)
Comment (redouan-rhazouani) [reply usecases/backup/scheduler.go:134]: Done

**Pull request merged**